apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -26,12 +26,38 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
-#define	STACK_NORMAL_SIZE		10
+#define	STACK_NORMAL_SIZE		300
 #define	STACK_OVERFLOW_SIZE		4096
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+void stack_prot_overflow(int *val, char ch)
+{
+	char *ptr;
+	int i = 0;
+	char array_normal[STACK_NORMAL_SIZE];
+
```

Comment:
It is difficult to understand what is being done in below code. Can you please add some comments for each operation?

Reply:
added comments